agent/consul/gateways/controller_gateways.go

```diff
@@ -840,151 +829,132 @@ func newGatewayMeta(gateway *structs.APIGatewayConfigEntry, bound structs.Config
 	}).initialize()
 }
 
-// gatewayConditionGenerator is a simple struct used for isolating
-// the status conditions that we generate for our components
-type gatewayConditionGenerator struct {
```

Comment:
this struct was no longer needed because we encapsulated creating the conditions in the structs package along with validations